### PR TITLE
Make ezFoundation build on arm uwp

### PR DESCRIPTION
**Pull request by Ingrater**

### CODE REVIEW

File: Code/BuildSystem/CMake/ezUtilsCppFlags.cmake [hunk 1]
```diff
@@ -77,18 +77,29 @@ function(ez_set_build_flags_msvc TARGET_NAME)
...
+	if (CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND EZ_CMAKE_ARCHITECTURE_X86)
 		target_compile_options(${TARGET_NAME} PRIVATE "-msse4.1")
 	endif()
 	
+	set (LINKER_FLAGS_DEBUG "")
+	
+	#Do not remove unreferenced data. Required to make incremental linking work.
+	set (LINKER_FLAGS_DEBUG "${LINKER_FLAGS_DEBUG} /OPT:NOREF")
+	
+	#Do not enable comdat foliding in debug. Required to make incremental linking work.
```
Comment (jankrassnigg) [inline Code/BuildSystem/CMake/ezUtilsCppFlags.cmake]: typo: foliding

File: Code/BuildSystem/CMake/ezUtilsCppFlags.cmake [hunk 2]
```diff
@@ -77,18 +77,29 @@ function(ez_set_build_flags_msvc TARGET_NAME)
...
+	
 	set (LINKER_FLAGS_RELEASE "")
 	
 	set (LINKER_FLAGS_RELEASE "${LINKER_FLAGS_RELEASE} /INCREMENTAL:NO")
 		
 	# Remove unreferenced data (does not work together with incremental build)
 	set (LINKER_FLAGS_RELEASE "${LINKER_FLAGS_RELEASE} /OPT:REF")
 		
-	# Don't know what it does, but Clemens wants it :-) (does not work together with incremental build)
+	# Enable comdat folding. Reduces the number of redudant template functions and thus reduces binary size. Makes debugging ahrder though.
```
Comment (jankrassnigg) [inline Code/BuildSystem/CMake/ezUtilsCppFlags.cmake]: typo: hrader

File: Code/Tools/Player/CMakeLists.txt [hunk 3]
```diff
@@ -25,5 +25,7 @@ endif()
 
 # set all external projects as runtime dependencies of this application
 get_property(EXTERNAL_PROJECTS GLOBAL PROPERTY "EXTERNAL_PROJECTS")
-message(STATUS "External Projects: ${EXTERNAL_PROJECTS}")
-add_dependencies(${PROJECT_NAME} ${EXTERNAL_PROJECTS})
+if(${EXTERNAL_PROJECTS})
```
Comment (jankrassnigg) [inline Code/Tools/Player/CMakeLists.txt]: afaik you can just write "if (EXTERNAL_PROJECTS)"

File: Code/Engine/Foundation/Math/Implementation/Math_inl.h [hunk 4]
```diff
@@ -109,13 +109,14 @@ namespace ezMath
 
   EZ_ALWAYS_INLINE ezUInt32 CountBits(ezUInt32 value)
   {
-#if EZ_ENABLED(EZ_PLATFORM_WINDOWS)
+#if EZ_ENABLED(EZ_PLATFORM_WINDOWS) && EZ_ENABLED(EZ_PLATFORM_ARCH_X86)
     return __popcnt(value);
```
Comment (C-Core) [inline Code/Engine/Foundation/Math/Implementation/Math_inl.h]: Should use _CountOneBits on arm

**Pull request merged**